### ISSUE TITLE
Dreambooth stops training after some steps without saying nothing

### ISSUE THREAD

**Issue by andreacostigliolo** [d8ahazard/sd_dreambooth_extension#1249]
## ⚠️If you do not follow the template, your issue may be closed without a response ⚠️
Kindly read and fill this form in its entirety.

### 0. Initial troubleshooting
Please check each of these before opening an issue. If you've checked them, **delete this section of your bug report**. Have you:
- Updated the Stable-Diffusion-WebUI to the latest version? Yes
- Updated Dreambooth to the latest revision? Yes
- Completely restarted the stable-diffusion-webUI, not just reloaded the UI? Yes 
- Read the [Readme](https://github.com/d8ahazard/sd_dreambooth_extension#readme)? Yes

### 1. Please find the following lines in the console and paste them below.

```
#######################################################################################################
Initializing Dreambooth
Dreambooth revision: 926ae204ef5de17efca2059c334b6098492a0641
Successfully installed google-auth-oauthlib-0.4.6

[+] xformers version 0.0.17 installed.
[+] torch version 2.0.1+cu118 installed.
[+] torchvision version 0.15.2+cu118 installed.
[+] accelerate version 0.18.0 installed.
[+] diffusers version 0.14.0 installed.
[+] transformers version 4.26.1 installed.
[+] bitsandbytes version 0.35.4 installed.

Launching Web UI with arguments: --xformers


#######################################################################################################
```

### 2. Describe the bug

Dreambooth stops training after some steps without saying nothing. Just "press a key to continue" and then it exits. Everytime with a different percentage (sometime 10% sometime 90%) using same parameters.  Sometimes it complete the training (1 out of 20).

**Screenshots/Config**
{
    "adamw_weight_decay": 0.01,
    "adaptation_beta1": 0,
    "adaptation_beta2": 0,
    "adaptation_d0": 1e-08,
    "adaptation_eps": 1e-08,
    "attention": "xformers",
    "cache_latents": false,
    "clip_skip": 1,
    "concepts_list": [
        {
            "class_data_dir": "",
            "class_guidance_scale": 7.5,
            "class_infer_steps": 40,
            "class_negative_prompt": "",
            "class_prompt": "",
            "class_token": "in the style of NealAdams",
            "instance_data_dir": "G:\\Download\\Neal Adams 512",
            "instance_prompt": "",
            "instance_token": "NealAdams",
            "is_valid": true,
            "n_save_sample": 1,
            "num_class_images_per": 0,
            "sample_seed": -1,
            "save_guidance_scale": 7.5,
            "save_infer_steps": 20,
            "save_sample_negative_prompt": "",
            "save_sample_prompt": "[filewords]",
            "save_sample_template": ""
        }
    ],
    "concepts_path": "",
    "custom_model_name": "",
    "noise_scheduler": "DDPM",
    "deterministic": false,
    "ema_predict": false,
    "epoch": 0,
    "epoch_pause_frequency": 0,
    "epoch_pause_time": 0,
    "freeze_clip_normalization": false,
    "gradient_accumulation_steps": 1,
    "gradient_checkpointing": false,
    "gradient_set_to_none": true,
    "graph_smoothing": 50,
    "half_model": false,
    "train_unfrozen": false,
    "has_ema": false,
    "hflip": false,
    "infer_ema": false,
    "initial_revision": 0,
    "learning_rate": 2e-06,
    "learning_rate_min": 1e-06,
    "lifetime_revision": 0,
    "lora_learning_rate": 0.0001,
    "lora_model_name": "",
    "lora_unet_rank": 4,
    "lora_txt_rank": 4,
    "lora_txt_learning_rate": 5e-05,
    "lora_txt_weight": 1,
    "lora_weight": 1,
    "lr_cycles": 1,
    "lr_factor": 0.5,
    "lr_power": 1,
    "lr_scale_pos": 0.5,
    "lr_scheduler": "constant_with_warmup",
    "lr_warmup_steps": 0,
    "max_token_length": 75,
    "mixed_precision": "bf16",
    "model_name": "SDNealAdams",
    "model_dir": "D:\\stable-diffusion-webui\\models\\dreambooth\\SDNealAdams",
    "model_path": "D:\\stable-diffusion-webui\\models\\dreambooth\\SDNealAdams",
    "num_train_epochs": 100,
    "offset_noise": 0,
    "optimizer": "8bit AdamW",
    "pad_tokens": true,
    "pretrained_model_name_or_path": "D:\\stable-diffusion-webui\\models\\dreambooth\\SDNealAdams\\working",
    "pretrained_vae_name_or_path": "",
    "prior_loss_scale": false,
    "prior_loss_target": 100.0,
    "prior_loss_weight": 0.75,
    "prior_loss_weight_min": 0.1,
    "resolution": 512,
    "revision": 0,
    "sample_batch_size": 1,
    "sanity_prompt": "",
    "sanity_seed": 420420.0,
    "save_ckpt_after": true,
    "save_ckpt_cancel": false,
    "save_ckpt_during": false,
    "save_ema": true,
    "save_embedding_every": 25,
    "save_lora_after": true,
    "save_lora_cancel": false,
    "save_lora_during": false,
    "save_lora_for_extra_net": false,
    "save_preview_every": 5,
    "save_safetensors": true,
    "save_state_after": false,
    "save_state_cancel": false,
    "save_state_during": false,
    "scheduler": "DEISMultistep",
    "shuffle_tags": true,
    "snapshot": "",
    "split_loss": true,
    "src": "D:\\stable-diffusion-webui\\models\\Stable-diffusion\\v1-5-pruned-emaonly.safetensors",
    "stop_text_encoder": 0.75,
    "strict_tokens": false,
    "tf32_enable": false,
    "train_batch_size": 2,
    "train_imagic": false,
    "train_unet": true,
    "use_concepts": false,
    "use_ema": false,
    "use_lora": true,
    "use_lora_extended": false,
    "use_subdir": true,
    "v2": false
}
### 3. Provide logs

If a crash has occurred, please provide the *entire* stack trace from the log, including the last few log messages *before* the crash occurred.

No error just stop and exit.

### 4. Environment

What OS? Windows

If Windows - WSL or native? native

What GPU are you using? 4070

**Comment by swumagic** [comment]
After each update, the requirements.txt must be modified as follows.   
extensions\sd_dreambooth_extension\requirements.txt。it run ok。

accelerate~=0.18.0
bitsandbytes==0.35.4
dadaptation~=3.1
diffusers~=0.16.1
discord-webhook~=1.1.0
fastapi~=0.94.1
gitpython~=3.1.31
lion-pytorch~=0.0.7
Pillow~=9.5.0
tqdm~=4.64.1
transformers~=4.25.1 # > 4.26.x causes issues (db extension #1110)

**Comment by andreacostigliolo** [comment]
But with this config, after unistall -r and install -r,  at the start I have this error : 

Initializing Dreambooth
Dreambooth revision: 926ae204ef5de17efca2059c334b6098492a0641
Successfully installed google-auth-oauthlib-0.4.6 transformers-4.25.1

[+] xformers version 0.0.17 installed.
[+] torch version 2.0.1+cu118 installed.
[+] torchvision version 0.15.2+cu118 installed.
[+] accelerate version 0.18.0 installed.
[+] diffusers version 0.16.1 installed.
[!] transformers NOT installed.
[+] bitsandbytes version 0.35.4 installed.

Successfully installed google-auth-oauthlib-0.4.6 transformers-4.25.1
and [!] transformers NOT installed.

But it seems to work even if there is that error.

**Comment by ArrowM** [comment]
@andreacostigliolo A1111 commits with "new" gradio (above version 16) have some issue with the ui timing out, but the training should continue and you should be able to see it in the console. Plz lmk if it doesn't.
As for the install errors, there should be more details in the console from where it is installing library if anything fails to install. What startup flags are you using?

@swumagic this is just wrong.

**Comment by andreacostigliolo** [comment]
Well it gives error but works. It's strange. I used clean install of both new versions, without touching anything except --Xformers in starting arguments.
